contracts/RolesRegistry/SftRolesRegistry.sol

```diff
@@ -0,0 +1,241 @@
+// SPDX-License-Identifier: CC0-1.0
+
+pragma solidity 0.8.9;
+
+import { IERCXXXX } from './interfaces/IERCXXXX.sol';
+import { IERC165 } from '@openzeppelin/contracts/utils/introspection/IERC165.sol';
+import { IERC1155 } from '@openzeppelin/contracts/token/ERC1155/IERC1155.sol';
+import { IERC1155Receiver } from '@openzeppelin/contracts/token/ERC1155/IERC1155Receiver.sol';
+import { ERC1155Holder, ERC1155Receiver } from '@openzeppelin/contracts/token/ERC1155/utils/ERC1155Holder.sol';
+import { ERC165Checker } from '@openzeppelin/contracts/utils/introspection/ERC165Checker.sol';
+
+// Semi-fungible token (SFT) registry with only one role (UNIQUE_ROLE)
+contract SftRolesRegistrySingleRole is IERCXXXX, ERC1155Holder {
+    bytes32 public constant UNIQUE_ROLE = keccak256('UNIQUE_ROLE');
+
+    // grantor => tokenAddress => operator => isApproved
+    mapping(address => mapping(address => mapping(address => bool))) public tokenApprovals;
+
+    // nonce => DepositInfo
+    mapping(uint256 => DepositInfo) public deposits;
+
+    // nonce => RoleAssignment
+    mapping(uint256 => RoleData) internal roleAssignments;
+
+    modifier validGrantRoleData(
+        uint256 _nonce,
+        uint64 _expirationDate,
+        uint256 _tokenAmount
+    ) {
+        require(_nonce > 0, 'SftRolesRegistry: nonce must be greater than zero');
+        require(_expirationDate > block.timestamp, 'SftRolesRegistry: expiration date must be in the future');
+        require(_tokenAmount > 0, 'SftRolesRegistry: tokenAmount must be greater than zero');
+        _;
+    }
+
+    modifier onlyOwnerOrApproved(address _account, address _tokenAddress) {
+        require(
+            _account == msg.sender || isRoleApprovedForAll(_tokenAddress, _account, msg.sender),
+            'SftRolesRegistry: account not approved'
+        );
+        _;
+    }
+
+    /** External Functions **/
+
+    function grantRoleFrom(
```

Comment:
Missing check to see if role is `UNIQUE_ROLE`

Reply:
it does not need to, the _role field is not being used